### PR TITLE
WIP: Provide current data from db for preUpdate hooks

### PR DESCRIPTION
**Pull request by nihaux**

### CODE REVIEW

**Comment by GromNaN** [comment]
Does it add a request systematically when the preUpdade hook is used or is it only when the previous data is expected?

**Comment by nihaux** [comment]
As it is, it loads the previous data when a preUpdate hook is used at least once in the resource.

File: packages/graphql-platform-core/src/graphql-platform/operation/mutation/update-one.ts [hunk 1]
```diff
@@ -719,6 +720,17 @@ export class UpdateOneOperation extends AbstractOperation<UpdateOneOperationArgs
     const update = new UpdateOneValue(resource, await this.parseDataComponentMap(args.data, context));
 
     if (resource.hasPreHook(ResourceHookKind.PreUpdate)) {
+      const selectionNode =
+          this.resource
+            .getFieldSet()
+            .filter(field => field.isPublic())
```
[the remark this answers is not in the document]
Comment (nihaux) [reply packages/graphql-platform-core/src/graphql-platform/operation/mutation/update-one.ts]: problem is, [you cannot select private fields for output](https://github.com/prismamedia/graphql-platform/blob/master/packages/graphql-platform-core/src/graphql-platform/resource/component/set.ts#L17)

File: packages/graphql-platform-core/src/graphql-platform/operation/mutation/update-one.ts [hunk 2]
```diff
@@ -734,6 +746,7 @@ export class UpdateOneOperation extends AbstractOperation<UpdateOneOperationArgs
                 resource,
                 field,
                 toBeUpdatedNodeId: nodeId,
+                toBeUpdatedNodeData: data,
```
[the remark this answers is not in the document]
Comment (nihaux) [reply packages/graphql-platform-core/src/graphql-platform/operation/mutation/update-one.ts]: done

**Pull request merged**